WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Sniff to check theme for deregister and register of core scripts (jquery)
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_NoDeregisterCoreScriptSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Core scripts to sniff for.
+	 *
+	 * @var array
+	 */
+	private $core_scripts = array(
+		'jquery',
```

Comment:
In WPCS they generally use a pattern like the below for these kind of arrays which allows for checking the array using `isset()` instead of `in_array()` (faster).

```
array(
	'jquery' => true,
)